docs/architecture/adr-020-protobuf-transaction-encoding.md

```diff
@@ -93,22 +117,19 @@ attempt to upstream important improvements to `Tx`.
 
 ### Signing
 
-Signatures are structured using the `SignDoc` below which reuses `TxBody` and only
-adds the fields which are needed for signatures but not present on `TxBody`:
+All of the signing modes below aim to provide the following guarantees:
 
-```proto
-// types/types.proto
-message SignDoc {
-    TxBody body = 1;
-    string chain_id = 2;
-    uint64 account_number = 3;
-    uint64 account_sequence = 4;
-}
-```
+* **No Malleability**: given a `TxBody` and an `AuthInfo` there is one and only
+one valid `Tx` that can be generated
+* **Predictable Gas**: if I am signing a transaction where I am paying a fee,
+the final gas is fully dependent on what I am signing
```

Comment:
Can you help me understand this point better: do we (right now and in the future) support an undefined number of signatures?

If not, the only purpose of this added complexity is to set different gas prices for different verification modes?

Reply:
As I understand it, with multisigs currently it could be unpredictable because the signatures are nested. Also, I am noting this in contrast to weave where my reading is that it's generally unpredictable.